### ISSUE TITLE
Introduce a `Value_Type.Null`.

### ISSUE THREAD

**Issue by radeusgd** [enso-org/enso#6281]
This is orthogonal to #5872.

Initially, I thought we can live without `Value_Type.Null`. Each of columns coming from the DB has some proper type. 

However, there are cases where this becomes problematic.

When I do `Table.new [["X", [Nothing]]]` - what type `X` should have? I guess for in-memory it's ok if it's mixed by default and we can cast it to any nullable type we want, and all will work OK.

However, there is the case of creating 'pure NULL' column in Expressions, both in-memory and in-DB.
Currently the behaviour is as follows:
```
t = table_builder [["X", [1, 2, 3]]]
t.compute "not(Nothing)" . should_fail_with Invalid_Value_Type
t.compute "Nothing + Nothing" . should_fail_with Illegal_Argument
t.compute "Nothing * Nothing" . should_fail_with Invalid_Value_Type
```

So any expression on nulls fails with Invalid Value Type, that's because the value is `Mixed` (in-memory) or `Unsupported_Data_Type "NULL"` (in-DB, at least in Postgres, I'm not even checking SQLite 🤕).

I'm not sure if that behaviour is right. It feels that if `t.at "X" + Nothing` works OK (if `X` is itself a numeric column), probably `[X] + Nothing` but also `Nothing + [X]` should work too.

The sane way to make it work that comes to my mind is only by adding a `Value_Type.Null` which these all-null columns can take. Such a special type can then be accepted by any `Value_Type.expect_*` operation, making these operations work.

- [ ] Add the `Null` type. It should not be mapped to-from SQL by default, I think.
- [ ] Make sure all `expect_*` functions also accept the `Null` type.
	- Should `is_*` answer True to it too? Figure out, initial thought is that probably not.
- [ ] Add tests for handling of these NULL columns in expressions.
- [ ] Update tests in `Aggregate_Column_Spec.enso`.

**Comment by radeusgd** [comment]
I'm not absolutely certain if we need this. If we want to keep the current behavior and accept the errors on Null in these cases, we can close this issue. But I think ideally, we should at some point make sure that expressions and raw column operations are more consistent. So I'd be for implementing this, but probably with a lower priority.

cc: @jdunkerley what do you think about this? I'd love to hear your opinion.

**Comment by jdunkerley** [comment]
Yes we should have a `Value_Type.Null` it is useful as can be "coerced" to whatever is needed.
Expressions needed this.

**Comment by radeusgd** [comment]
- [ ] We should also update tests that were disabled until this is fixed.
- [ ] In particular, we want to update the Union type unification logic so that Null columns are ignored when computing the resulting type (e.g. Null+Integer->Null, not Mixed), only if _all_ columns that are being merged are Null then the result is Null.
